### DISCUSSION TITLE
Add TryParseSvgPath functionality

### DISCUSSION THREAD

**Discussion by JimBobSquarePants** [SixLabors/ImageSharp.Drawing#205] [Ideas] [answered]
I think it would be an invaluable feature for our `PathBuilder` type to be able to parse an SVG paths.

We could add a method with the following signature.

```c#
bool TryParseSvgPath(ReadonlySpan<char> path, out bool value);
```

We can use the Skia `SkPath.FromSVGString` method as inspiration.

https://github.com/google/skia/blob/5deac304821b04176c0c3ac9535e5889a2966265/src/utils/SkParsePath.cpp#L75-L203

Note. We follow the System.Drawing API model for our `PathBuilder` type. Skia uses a different model that includes `MoveTo` `LineTo` methods so we would need to translate that across.

**Comment by tocsoft** [comment, marked as answer]
FYI the `BaseGlyphBuilder` works in that same builder model with the `MoveTo`/`LineTo` style so that code could probably be hoisted into a shared component/PathBuilder wrapper type that exposes that API publicly and we would just utilise it in the glyph builder rather than implementing it there directly.

**Reply by JimBobSquarePants** [reply]
Interesting.... Would you be able to knock out a quick outline of the API you have in mind?

**Reply by tocsoft** [reply]
https://github.com/SixLabors/ImageSharp.Drawing/pull/206 should do it... we probably have some missing api/cases as those are just the ones we can utilise for GlyphRendering but we can expand as required.

That's the class I would hand the SVG path parsing logic off too (at least as an extension method).

**Reply by JimBobSquarePants** [reply]
Thanks, will have a proper look. This will be super useful for drawing text along a path.

**Reply by JimBobSquarePants** [reply]
@tocsoft I'll finish off your branch and do a PR now that we have `ArcTo`.

**Reply by JimBobSquarePants** [reply]
Yep, Nope. I can't read that C++ it's nightmarish.

**Reply by tocsoft** [reply]
#217 is a mostly working version... has an outstanding issue on `T` commands https://svgwg.org/specs/paths/#PathDataQuadraticBezierCommands but its actually parsing/compiling with a few test cases i've lifted from the SVG specs themselves.
